### PR TITLE
remove containers after service is destroyed

### PR DESCRIPTION
**Pull request by earthquakesan**
same behaviour as with docker swarm update --task-history-limit=0

### CODE REVIEW

File: platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java [hunk 1]
```diff
@@ -587,16 +587,19 @@ public String startContainer(String imageName, String containerType, String pare
...
+            Task taskInfo = dockerClient.inspectTask(taskId);
+            String containerId = taskInfo.status().containerStatus().containerId();
             String serviceId = taskInfo.serviceId();
 
             // If we are not in testing mode, remove all containers. In testing
             // mode, remove only those that have a non-zero status
             Integer exitCode = taskInfo.status().containerStatus().exitCode();
             if (!DEPLOY_ENV.equals(DEPLOY_ENV_DEVELOP) && ((!DEPLOY_ENV.equals(DEPLOY_ENV_TESTING)) || (exitCode == null) || (exitCode == 0))) {
                 dockerClient.removeService(serviceId);
+                dockerClient.stopContainer(containerId, 10);
```
Comment (MichaelRoeder) [inline platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java]: Stopping the container(s) should be done in any case. Only removing them is based on the `if` above.

Comment (earthquakesan) [reply platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java]: What is the purpose of the if statement there then? Shall I remove it?

File: platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java [hunk 2]
```diff
@@ -587,16 +587,19 @@ public String startContainer(String imageName, String containerType, String pare
 
 
     @Override
-    public void removeContainer(String containerId) {
+    public void removeContainer(String taskId) {
         try {
-            Task taskInfo = dockerClient.inspectTask(containerId);
+            Task taskInfo = dockerClient.inspectTask(taskId);
+            String containerId = taskInfo.status().containerStatus().containerId();
```
Comment (denkv) [inline platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java]: Can `containerId` be `null` there? Everything will be okay with this?

Comment (earthquakesan) [reply platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java]: Hmmm... It can, need to make a check for this.

Comment (earthquakesan) [reply platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java]: @denkv I changed the sequence, first removing container and then service, that would prevent the potential problem

**Pull request merged**